docs/xxportablesharedcache.md

```diff
@@ -24,9 +24,9 @@
 
 # -XX:\[+|-\]PortableSharedCache
 
-This option enables AOT compiled code to be generated based on a chosen set of processor features (features present in the Intel&reg; Sandy Bridge microarchitecture) that maximizes its portability across machines. It is currently supported only on x86. The feature is turned on by default in Docker containers but can be disabled with `-XX:-PortableSharedCache`. To enable the portable shared cache feature outside Docker containers, `-XX:+PortableSharedCache` needs to be specified for the initial Java Virtual Machine instance (when the creation of the Shared Class Cache happens) as well as for every subsequent instance that makes use of the same Shared Class Cache.
+This option enables AOT compiled code to be generated based on a chosen set of processor features (features present in the Intel&reg; Sandy Bridge microarchitecture) that maximizes its portability across processors of different microarchitectures (Intel&reg; Sandy Bridge or newer). This option also enables AOT compiled code to be generated in a way that is portable across OpenJ9 Compressed Reference builds with a heap size in the range of 0 GB to 28 GB. This portable AOT feature has a steady-state throughput penalty of 1% to 2% for OpenJ9 Compressed Reference builds with a heap size in the range of 0 GB to 3GB. It is turned on by default in Docker containers and can be disabled with `-XX:-PortableSharedCache`. It is currently supported only on x86. The feature is turned on by default in Docker containers but can be disabled with `-XX:-PortableSharedCache`. To enable the portable shared cache feature outside Docker containers, `-XX:+PortableSharedCache` needs to be specified for the initial Java Virtual Machine instance (when the creation of the Shared Class Cache happens) as well as for every subsequent instance that makes use of the same Shared Class Cache.
```

Comment:
"heap size in the range of 0 GB to 28 GB", technically speaking the smallest heap one can have is 4MB.

Reply:
"and can be disabled" --> "but can be disabled"

Reply:
Actually I see that that line is duplicated

Reply:
This option enables AOT compiled code to be generated based on a chosen set of processor features (features present in the Intel&reg; Sandy Bridge microarchitecture) that maximizes its portability across processors of different microarchitectures (Intel&reg; Sandy Bridge or newer). This option also enables AOT compiled code to be generated in a way that is portable across OpenJ9 Compressed Reference builds with a heap size in the range of 4 MB to 28 GB. This portable AOT feature may introduce a small (1-2%) steady-state throughput penalty when OpenJ9 Compressed Reference builds are used with a heap size in the range of 4 MB to 3GB. It is currently supported only on x86. The feature is turned on by default in Docker containers but can be disabled with `-XX:-PortableSharedCache`. To enable the portable shared cache feature outside Docker containers, `-XX:+PortableSharedCache` needs to be specified for the initial Java Virtual Machine instance (when the creation of the Shared Class Cache happens) as well as for every subsequent instance that makes use of the same Shared Class Cache.
This option is especially useful for applications deployed on the cloud in the form of Docker containers, as it allows the Shared Class Cache to be portable across processors with different microarchitectures and across Compressed Reference OpenJ9 builds with different heap sizes.

Reply:
I suggest the following rewrite of Marius's update with tweaks for consistency and clarity, including moving the platform restriction up front and making it similar to other topics such as https://www.eclipse.org/openj9/docs/xxhandlesigxfsz/, de-capitalising various terms, and using a more active voice. Copyright symbols are required only on the first instance. Also...  
- I changed "chosen set" to "particular" set otherwise you wonder who's doing the choosing :) 
- I assume we don't really mean builds, ie a build of OpenJDK that contains OpenJ9 , but rather instances of the VM that are using compressed refs 

-----

**(x86 platforms only)**

This option, when enabled, increases the portability of AOT-compiled code, in the following ways:
- The code is generated based on a particular set of processor features (present in the Intel® Sandy Bridge microarchitecture) that maximize its portability across processors of different microarchitectures (Intel Sandy Bridge or newer). 
- The code is generated so that it is portable across OpenJ9 VMs that use compressed references and have a heap size of 4 MB to 28 GB (previously, AOT-compiled code could not be shared between VMs that use compressed references and that have different heap sizes). This feature might introduce a small (1-2%) steady-state throughput penalty when compressed references are used and the heap size is between 4 MB and 3GB. 

This option is especially useful for applications that are deployed on the cloud in the form of Docker containers because it allows the shared classes cache to be portable across processors with different microarchitectures and across compressed reference VMs with different heap sizes. 

This option is therefore enabled by default in Docker containers but you can disable it by specifying `-XX:-PortableSharedCache` on the command line. To enable the portable shared cache feature outside Docker containers, specify `-XX:+PortableSharedCache` for the initial virtual machine instance (which is when the shared classes cache is created) and for every subsequent instance that uses the same shared classes cache.

Reply:
Sounds much better. Thanks @doveye !